File: packages/gbnf-experiment/data/20260909T040730Z_0947f229/ported_implementation/gbnf/grammar_graph/colorize.py
"""Port of ``src/grammar-graph/colorize.ts``."""

from __future__ import annotations

from enum import Enum
from typing import Union


class Color(str, Enum):
    BLUE = '\x1b[34m'
    CYAN = '\x1b[36m'
    GREEN = '\x1b[32m'
    RED = '\x1b[31m'
    GRAY = '\x1b[90m'
    YELLOW = '\x1b[33m'


def colorize(value: Union[str, int], color: Color) -> str:
    return f'{color.value}{value}'


def no_color(value: Union[str, int], color: Color) -> str:
    return f'{value}'

File: proposed-projects/round-trip-experiment/reverse/20260909T190508Z_7c1544c0/20260910T103012Z_7e3d5183/ported_implementation/src/grammar-graph/colorize.ts
export enum Color {
  BLUE = '\x1b[34m',
  CYAN = '\x1b[36m',
  GREEN = '\x1b[32m',
  RED = '\x1b[31m',
  GRAY = '\x1b[90m',
  YELLOW = '\x1b[33m',
}

export type Colorize = (string: string | number, color: Color) => string;

export const colorize: Colorize = (string, color) => `${color}${string}`;

export const noColor: Colorize = (string) => `${string}`;
